src/index.ts

```diff
@@ -110,20 +120,18 @@ export const replacePathnameInUrl = flipCurried(
     ),
 );
 
-const appendPathnameToParsedUrl = ({
-    pathnameToAppend,
-}: {
-    pathnameToAppend: string;
-}): MapUrlFn => ({ parsedUrl }) => {
-    const pathnameParts = pipe(
-        () => mapMaybe(parsedUrl.pathname, getPartsFromPathname),
-        maybe => getOrElseMaybe(maybe, () => []),
-    )({});
-    const pathnamePartsToAppend = getPartsFromPathname(pathnameToAppend);
-    const newPathnameParts = [...pathnameParts, ...pathnamePartsToAppend];
-    const newPathname = getPathnameFromParts(newPathnameParts);
-    return { ...parsedUrl, pathname: newPathname };
-};
+const appendPathnameToParsedUrl = ({ pathnameToAppend }: { pathnameToAppend: string }): MapUrlFn =>
+    replacePathnameInParsedUrl({
```

Comment:
Realised we can implement add/append functions in terms of the replace functions. The other changes in this PR are just re-orders.

In order for this to work, the replace functions must accept an "update function" and provide the previous value to that. This is just like how lenses work:

```ts
import { Lens } from 'monocle-ts';
import { UrlWithStringQuery } from 'url';

const pathnameLens = Lens.fromProp<UrlWithStringQuery>()('pathname');
const replacePathnameInUrl = pathnameLens.modify;
```